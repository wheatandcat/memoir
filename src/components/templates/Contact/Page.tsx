import React, { memo, useCallback } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import * as Clipboard from 'expo-clipboard';
import { ConnectedType } from 'components/pages/Contact/Connected';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';
import { TextInput as RNTextInput } from 'react-native';
import useAutoFocusInput from 'hooks/useAutoFocusInput';

export type Props = ConnectedType & {};

const Page: React.FC<Props> = (props) => {
  const autoFocusProps = useAutoFocusInput(true, 500);

  const onCopyUserID = useCallback(() => {
    Clipboard.setString(props.userID);
  }, [props.userID]);

  const debug = false;

  return (
    <View style={styles.root}>
      {debug && (
        <>
          <View style={styles.title}>
            <Text color="secondaryLight" size="sm">
              ユーザーID
            </Text>
          </View>
          <View style={styles.userIDText}>
            <TouchableOpacity onPress={onCopyUserID}>
              <Text size="sm">{props.userID}</Text>
            </TouchableOpacity>
          </View>
        </>
      )}

      <View style={styles.title} pt={debug ? 4 : 0} pb={2}>
        <Text color="secondaryLight" size="sm">
          コメント
        </Text>
      </View>
      <RNTextInput
        multiline
        style={styles.inputText}
        autoCapitalize="none"
        onChangeText={props.onChangeText}
        maxLength={300}
        blurOnSubmit={false}
        {...autoFocusProps}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    paddingTop: theme().space(4),
  },
  title: {
    paddingVertical: theme().space(1),
    paddingHorizontal: theme().space(3),
  },
  userIDText: {
    backgroundColor: theme().color.background.light,
    paddingVertical: theme().space(3),
    paddingHorizontal: theme().space(3),
    borderTopWidth: StyleSheet.hairlineWidth,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderColor: theme().color.secondary.light,
  },
  inputText: {
    borderColor: theme().color.secondary.light,
    borderTopWidth: StyleSheet.hairlineWidth,
    borderBottomWidth: StyleSheet.hairlineWidth,
    paddingTop: theme().space(2),
    paddingBottom: theme().space(1),
    paddingHorizontal: theme().space(3),
    height: 250,
    color: theme().color.secondary.main,
    backgroundColor: theme().color.background.light,
    lineHeight: 20,
    fontWeight: 'bold',
    textAlignVertical: 'top',
  },
});

export default memo(Page);
