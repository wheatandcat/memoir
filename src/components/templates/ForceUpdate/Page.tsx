import React, { memo, useCallback } from 'react';
import { StyleSheet } from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';
import AppLink from 'react-native-app-link';
import Button from 'components/atoms/Button';

export type Props = {};
const appStoreId = 1613532672;
const playStoreId = 'com.wheatandcat.memoir';

const Page: React.FC<Props> = () => {
  const onPress = useCallback(() => {
    AppLink.openInStore({
      appName: 'memoir',
      appStoreId,
      appStoreLocale: 'jp',
      playStoreId,
    });
  }, []);

  return (
    <View style={styles.root}>
      <View py={5} mx={3}>
        <Text lineHeight={30}>
          新しいバージョンがリリースされました。{'\n'}
          ストアからアプリのバージョンアップをして下さい。
        </Text>
      </View>

      <Button title="ストアへ" width={250} onPress={onPress} />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    padding: theme().space(3),
    height: '100%',
    alignItems: 'center',
  },
});

export default memo(Page);
