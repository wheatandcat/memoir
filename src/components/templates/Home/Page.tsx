import React, { memo, useState, useCallback } from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import View from 'components/atoms/View';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import 'dayjs/locale/ja';
import DateInput from 'components/organisms/DateInput/DateInput';
import MemoirButton from 'components/molecules/Home/MemoirButton.tsx';
import Cards from 'components/organisms/Cards/Cards';
import SettingModal from 'components/organisms/SettingModal';
import AddItemModal from 'components/organisms/AddItemModal';
import { LinearGradient } from 'expo-linear-gradient';
import theme from 'config/theme';

dayjs.locale('ja');
dayjs.extend(advancedFormat);

type Props = {
  date: string;
  openSettingModal: boolean;
  onItem: () => void;
  onMemoir: () => void;
  onCloseSettingModal: () => void;
  onChangeDate: (date: string) => void;
};

type State = {
  openAddItemModal: boolean;
};

const initialState = (): State => ({
  openAddItemModal: false,
});

const Page: React.FC<Props> = (props) => {
  const [state, setState] = useState<State>(initialState());

  const onAddItem = useCallback(() => {}, []);

  return (
    <LinearGradient
      // Background Linear Gradient
      colors={[theme().color.gradation[0], theme().color.gradation[1]]}
      style={styles.root}
    >
      <SettingModal
        isVisible={props.openSettingModal}
        onClose={props.onCloseSettingModal}
      />
      <AddItemModal
        isVisible={state.openAddItemModal}
        date={dayjs().format('YYYY-MM-DD')}
        onAdd={onAddItem}
        onClose={() => setState((s) => ({ ...s, openAddItemModal: false }))}
      />
      <ScrollView>
        <View style={styles.inner}>
          <DateInput date={props.date} onChange={props.onChangeDate} />
          <Cards
            onItem={props.onItem}
            onAddItem={() =>
              setState((s) => ({ ...s, openAddItemModal: true }))
            }
          />
        </View>
      </ScrollView>
      <MemoirButton onPress={props.onMemoir} />
    </LinearGradient>
  );
};

export default memo(Page);

const styles = StyleSheet.create({
  root: {
    height: '100%',
  },
  inner: {
    height: '100%',
  },
});
