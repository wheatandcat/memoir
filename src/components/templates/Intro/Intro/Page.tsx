import React, { memo } from 'react';
import { ConnectedType } from 'components/pages/Intro/Intro/Connected';
import IntroPanel from 'components/organisms/Intro/Intro';

export type Props = ConnectedType & {
  dayOfWeek: number;
  hours: number;
  minutes: number;
  notification: boolean;
};

const Intro: React.FC<Props> = (props) => {
  return (
    <IntroPanel
      dayOfWeek={props.dayOfWeek}
      hours={props.hours}
      minutes={props.minutes}
      notification={props.notification}
      onSaveNotification={props.onSaveNotification}
      onFinish={props.onFinish}
    />
  );
};

export default memo(Intro);
