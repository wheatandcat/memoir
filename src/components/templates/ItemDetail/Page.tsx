import React, { memo } from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import View from 'components/atoms/View';
import CardDetail from 'components/organisms/CardDetail/CardDetail';
import InputDateWrap from 'components/organisms/InputDateWrap/InputDateWrap';
import theme from 'config/theme';
import { ItemQuery } from 'queries/api/index';

type Item = NonNullable<ItemQuery['item']>;

type Props = Pick<Item, 'categoryID' | 'title'> & {
  loading: boolean;
  date: string;
  onChangeDate: (date: string) => void;
};

const Page: React.FC<Props> = (props) => {
  return (
    <InputDateWrap
      date={props.date}
      onChangeDate={props.onChangeDate}
      firstItem
    >
      <ScrollView>
        <View style={styles.inner}>
          {props.loading && (
            <CardDetail
              date={props.date}
              title={props.title}
              categoryID={props.categoryID}
            />
          )}
        </View>
      </ScrollView>
    </InputDateWrap>
  );
};

export default memo(Page);

const styles = StyleSheet.create({
  inner: {
    height: '100%',
    marginHorizontal: theme().space(3),
    marginVertical: theme().space(4),
  },
});
