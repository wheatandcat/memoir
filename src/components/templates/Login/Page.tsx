import React, { memo } from 'react';
import { StyleSheet } from 'react-native';
import theme from 'config/theme';
import View from 'components/atoms/View';
import { UseFirebaseAuth } from 'hooks/useFirebaseAuth';
import Form from 'components/organisms/Login/Form';

export type Props = {
  onAppleLogin: UseFirebaseAuth['onAppleLogin'];
  onGoogleLogin: UseFirebaseAuth['onGoogleLogin'];
};

const Page: React.FC<Props> = (props) => {
  return (
    <View style={styles.root}>
      <Form
        onAppleLogin={props.onAppleLogin}
        onGoogleLogin={props.onGoogleLogin}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    alignItems: 'center',
    padding: theme().space(4),
  },
});

export default memo(Page);
