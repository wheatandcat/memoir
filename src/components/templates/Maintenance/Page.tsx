import React, { memo, useEffect } from 'react';
import { StyleSheet } from 'react-native';
import { AppConfig } from 'lib/appConfig';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';

export type Props = AppConfig & {
  getMaintenance: () => void;
};

const Page: React.FC<Props> = (props) => {
  useEffect(() => {
    const interval = setInterval(() => {
      props.getMaintenance();
    }, 30000);

    return () => clearInterval(interval);
  }, [props]);

  return (
    <View style={styles.root}>
      <View py={5}>
        <Text textAlign="center">ただいまメンテンナンス中です。</Text>
      </View>

      {!!props.maintenancePeriod && (
        <View px={3}>
          <Text textAlign="center" color="secondaryLight" lineHeight={36}>
            【メンテンナンス日程】
          </Text>

          <Text textAlign="center" color="error">
            {props.maintenancePeriod}
          </Text>
        </View>
      )}

      <View px={3} pt={4}>
        <Text variants="small" color="secondaryLight">
          {props.maintenanceMessage}
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    alignItems: 'center',
  },
});

export default memo(Page);
