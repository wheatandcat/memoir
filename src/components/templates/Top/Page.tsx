import React, { memo } from 'react';
import { StyleSheet, ImageBackground, TouchableOpacity } from 'react-native';
import { ConnectedType } from 'components/pages/Top/Connected';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';
import Form, { Props as FormProps } from 'components/organisms/Login/Form';

export type Props = ConnectedType & FormProps & {};

const Page: React.FC<Props> = (props) => {
  return (
    <View style={styles.root}>
      <ImageBackground
        source={require('../../../img/common/frame.png')}
        resizeMode="cover"
        style={styles.image}
      >
        <View style={styles.title}>
          <Text textAlign="center" variants="logo">
            memoir
          </Text>
          <View style={styles.skip}>
            <TouchableOpacity onPress={props.onSkip}>
              <Text
                variants="small"
                style={styles.skipText}
                fontFamily="NotoSansJP-Bold"
                underline
              >
                スキップ
              </Text>
            </TouchableOpacity>
          </View>
        </View>

        <View style={styles.contents}>
          <View style={styles.contentsTitle}>
            <View style={styles.divider} />
            <View>
              <Text variants="small">新規登録</Text>
            </View>
            <View style={styles.divider} />
          </View>
          <View style={styles.signed}>
            <Form
              onAppleLogin={props.onAppleLogin}
              onGoogleLogin={props.onGoogleLogin}
            />
          </View>
          <View style={styles.footer}>
            <View style={styles.login}>
              <Text variants="small">
                すでにアカウントアカウントをお持ちの方
              </Text>
            </View>
            <View style={styles.loginText}>
              <TouchableOpacity>
                <Text color="primary" fontFamily="NotoSansJP-Bold" underline>
                  ログイン
                </Text>
              </TouchableOpacity>
            </View>
          </View>
        </View>
      </ImageBackground>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
    position: 'relative',
  },
  image: {
    flex: 1,
    justifyContent: 'center',
  },
  skip: {
    position: 'absolute',
    top: theme().space(3),
    right: theme().space(3),
  },
  skipText: {
    textDecorationLine: 'underline',
  },
  title: {
    height: '40%',
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  contents: {
    height: '60%',
    width: '100%',
    backgroundColor: theme().color.background.light,
  },
  contentsTitle: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    height: 80,
  },
  divider: {
    width: '30%',
    borderWidth: StyleSheet.hairlineWidth,
    borderColor: theme().color.base.main,
    marginHorizontal: theme().space(4),
  },
  signed: {
    paddingTop: theme().space(3),
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  footer: {
    alignItems: 'flex-end',
    justifyContent: 'space-between',
    flexDirection: 'row',
    flex: 1,
    bottom: theme().space(3),
    paddingHorizontal: theme().space(3),
    paddingTop: theme().space(3),
  },
  login: {
    bottom: theme().space(1),
  },
  loginText: {},
});

export default memo(Page);
