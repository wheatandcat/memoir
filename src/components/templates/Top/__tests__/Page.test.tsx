import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Page, { Props } from '../Page';

const propsData = (): Props => ({
  onAppleLogin: jest.fn(),
  onGoogleLogin: jest.fn(),
  onSkip: jest.fn(),
});

describe('components/templates/Top/Page.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Page {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
