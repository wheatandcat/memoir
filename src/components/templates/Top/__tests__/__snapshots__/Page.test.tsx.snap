// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/templates/Top/Page.tsx 正常にrenderすること 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E3C95D",
      "height": "100%",
    }
  }
>
  <ExpoStatusBar
    backgroundColor="#E3C95D"
    style="dark"
  />
  <RNCSafeAreaView>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "height": "100%",
        }
      }
    >
      <ImageBackground
        resizeMode="cover"
        source={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "40%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            fontFamily="RobotoCondensed-Bold"
            textAlign="center"
            variants="logo"
          >
            memoir
          </Text>
          <View
            style={
              Object {
                "position": "absolute",
                "right": 16,
                "top": 16,
              }
            }
          >
            <ForwardRef
              onPress={[MockFunction]}
            >
              <Text
                fontFamily="NotoSansJP-Bold"
                style={
                  Object {
                    "textDecorationLine": "underline",
                  }
                }
                underline={true}
                variants="small"
              >
                スキップ
              </Text>
            </ForwardRef>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "height": "60%",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 80,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "#D8D7D6",
                  "borderWidth": 0.5,
                  "marginHorizontal": 32,
                  "width": "30%",
                }
              }
            />
            <View>
              <Text
                fontFamily="RobotoCondensed-Bold"
                variants="small"
              >
                新規登録
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderColor": "#D8D7D6",
                  "borderWidth": 0.5,
                  "marginHorizontal": 32,
                  "width": "30%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingTop": 16,
                "width": "100%",
              }
            }
          >
            <Memo(Form)
              onAppleLogin={[MockFunction]}
              onGoogleLogin={[MockFunction]}
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "bottom": 16,
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 16,
                "paddingTop": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "bottom": 4,
                }
              }
            >
              <Text
                fontFamily="RobotoCondensed-Bold"
                variants="small"
              >
                すでにアカウントアカウントをお持ちの方
              </Text>
            </View>
            <View
              style={Object {}}
            >
              <ForwardRef
                onPress={[MockFunction]}
              >
                <Text
                  color="primary"
                  fontFamily="NotoSansJP-Bold"
                  underline={true}
                  variants="body"
                >
                  ログイン
                </Text>
              </ForwardRef>
            </View>
          </View>
        </View>
      </ImageBackground>
    </View>
  </RNCSafeAreaView>
</View>
`;
